schemas/mod-schema.json

```diff
@@ -109,12 +121,18 @@
 					},
 					"dependencies": {
 						"type": "object",
+						"description": "Other mods/libraries that this mod uses",
 						"additionalProperties": {
 							"type": "object",
 							"required": [
 								"version"
 							],
 							"properties": {
+								"optional": {
+									"type": "boolean",
+									"default": false,
+									"description": "Marks this dependency as optional. When set, the mod can function without it but may provide additional features or compatibility when available"
+								},
 								"version": {
 									"type": "string"
```

Comment:
maybe a description and pattern constraint should be added to this version field and the conflicts version field